cvmfs/server/cvmfs_server_publish.sh

```diff
@@ -494,7 +494,7 @@ filter_auto_tags() {
     $(get_swissknife_proxy)                    \
     -x $(get_follow_http_redirects_flag)       | \
     grep -E \
-    '^generic(_[[:digit:]]+)?-[[:digit:]]{4}-[[:digit:]]{2}-[[:digit:]]{2}T[[:digit:]]{2}:[[:digit:]]{2}:[[:digit:]]{2}Z' | \
+    '^generic(_[[:digit:]]+)?-[[:digit:]]{4}-[[:digit:]]{2}-[[:digit:]]{2}T[[:digit:]]{2}:[[:digit:]]{2}:[[:digit:]]{2}\.?[[:digit:]]{0,3}Z' | \
```

Comment:
i think the added regex needs to be changed a bit to  `(\.[[:digit:]]{1,3})?`
```
^generic(_[[:digit:]]+)?-[[:digit:]]{4}-[[:digit:]]{2}-[[:digit:]]{2}T[[:digit:]]{2}:[[:digit:]]{2}:[[:digit:]]{2}(\.[[:digit:]]{1,3})?Z
```

in your case the dot (.) is not mandatory. so it is ok to have `2 digits <no dot> up to 3 digits` which creates at least 2 invalid cases:
```
generic_2-2023-05-07T15:05:56115Z // would be valid but should not
generic_2-2023-05-07T15:05:56.Z // would be valid but should not
```

the new version with the brackets should enforce that the entire construct `.{1-3 digits}` must be  there or not at all.